server_config_test.cc

```diff
@@ -22,81 +22,149 @@ class ServerConfigTest : public ::testing::Test {
+    std::string port = "";
+    int ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ("1234", server_config->propertyLookUp("port"))
+    EXPECT_EQ(0, ec)
+      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ("1234", port)
       << "A listen property with a port number in a doubly nested block \
           should yield a port number";
 }
 
 TEST_F(ServerConfigTest, NonPortPropertiesPresent) {
     std::unique_ptr<ServerConfig> server_config
       = ParseString("server { echo true; listen 1234; }");
+    std::vector<std::string> query = {"server", "listen"};
+    std::string port = "";
+    int ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ("1234", server_config->propertyLookUp("port"))
+    EXPECT_EQ(0, ec)
+      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ("1234", port)
       << "A listen property with a port number in a doubly nested block with \
           other properties should yield a port number";
 }
 
 TEST_F(ServerConfigTest, HighlyNestedPropertiesSet) {
     std::unique_ptr<ServerConfig> server_config
       = ParseString("outer { server { echo true; listen 1234; } }");
+    std::vector<std::string> query = {"outer", "server", "listen"};
+    std::string port = "";
+    int ec = server_config->propertyLookUp(query, port);
 
-    EXPECT_EQ("1234", server_config->propertyLookUp("port"))
+    EXPECT_EQ(0, ec)
+      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ("1234", port)
       << "A listen property with a port number in a triply nested block with \
           other properties should yield a port number";
 }
 
 TEST_F(ServerConfigTest, PropertiesSetWithNewlines) {
     std::unique_ptr<ServerConfig> server_config
       = ParseString("server { echo \ntrue; listen \n1234; }");
+    std::vector<std::string> query = {"server", "listen"};
+    std::string port = "";
+    int ec = server_config->propertyLookUp(query, port);
+
+    EXPECT_EQ(0, ec)
+      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ("1234", port)
+      << "A listen property with a port number in a nested block with \
+          other properties and newlines should yield a port number";
+}
 
-    EXPECT_EQ("1234", server_config->propertyLookUp("port"))
+TEST_F(ServerConfigTest, MultiWordProperties) {
+    std::unique_ptr<ServerConfig> server_config
+      = ParseString("server { \
+                       echo \ntrue; \
+                       listen \n1234; \
+                       location /echo { \
+                         action echo; \
+                       } \
+                     }");
+    std::vector<std::string> query = {"server", "location /echo", "action"};
+    std::string action = "";
+    int ec = server_config->propertyLookUp(query, action);
+
+    EXPECT_EQ(0, ec)
+      << "Error value of 0 should be returned if property is found";
+    EXPECT_EQ("echo", action)
```

Comment:
Addressed in [latest commit](https://github.com/UCLA-CS130/Mr.-Robot-et-al./pull/37/commits/550e6e06ce66854e5500a8ddd875219703b36bd2)